shared/modules/confirmation.utils.ts

```diff
@@ -0,0 +1,82 @@
+import { TransactionType } from '@metamask/transaction-controller';
+import { ApprovalType } from '@metamask/controller-utils';
+
+export const REDESIGN_SIGNATURE_APPROVAL_TYPES = [
+  ApprovalType.EthSignTypedData,
+  ApprovalType.PersonalSign,
+];
+
+const REDESIGN_USER_TRANSACTION_TYPES = [
+  TransactionType.contractInteraction,
+  TransactionType.deployContract,
+  TransactionType.tokenMethodApprove,
+  TransactionType.tokenMethodIncreaseAllowance,
+  TransactionType.tokenMethodSetApprovalForAll,
+  TransactionType.tokenMethodTransfer,
+  TransactionType.tokenMethodTransferFrom,
+  TransactionType.tokenMethodSafeTransferFrom,
+  TransactionType.simpleSend,
+];
+
+export const REDESIGN_DEV_TRANSACTION_TYPES = [
+  ...REDESIGN_USER_TRANSACTION_TYPES,
+];
+
+const isCorrectDeveloperTransactionType = (
```

Comment:
Minor, can we use `function` rather than `const` since these are simple utility functions?

Reply:
is there a reason to prefer function declaration rather than function expressions? 🤔 

I can see both being used on different files... if there's reason to prefer one over the other we should probably add a linting rule like https://eslint.org/docs/latest/rules/func-style#allowarrowfunctions
Well I would say that consistency should be key here... but it seems that that ship has sailed 😂

Reply:
I prefer `function` for utility files for the hoisting, so it doesn't matter what order you declare them in and so can define the exported functions at the top for visibility, and internal functions underneath. Plus the intent is clearer as they are conceptually simple functions.

I use `const` for conceptual things like React components and where the scope / `this` needs to be inherited.